web/src/features/charts/bar-breakdown/ProductionSourceLegend.tsx

```diff
@@ -0,0 +1,30 @@
+import { ElectricityModeType } from 'types';
+import { modeColor } from 'utils/constants';
+
+import { LABEL_MAX_WIDTH, PADDING_Y } from './constants';
+import ProductionSourceIcon from './ProductionsSourceIcons';
+
+export default function ProductionSourceLegend({
+  electricityType,
+}: {
+  electricityType: ElectricityModeType;
+}) {
+  return (
+    <g className="pointer-events-none">
+      <rect
+        transform={`translate(${LABEL_MAX_WIDTH - 1.5 * PADDING_Y - 10}, 0)`}
+        fill={modeColor[electricityType as ElectricityModeType]}
+        width={14}
+        height={14}
+        rx={2}
+      />
+      <g
+        transform={`translate(${LABEL_MAX_WIDTH - 1.5 * PADDING_Y - 7}, 3)`}
+        width={14}
+        height={8}
+      >
+        <ProductionSourceIcon source={electricityType} />
+      </g>
+    </g>
```

Comment:
Maybe we can still wait with updating the flags after all, as I really can't anticipate the direction of the placement of the icons in general. I'll explore a bit more and come back to you, but ideally it would make more sense to maybe left align everything and having the icons leading instead of trailing (this will also make it more consistent with the tooltips and the general direction). Hope that makes sense! 🙌